src/common/iamclient/tlscredentials.hpp

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2025 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef AOS_COMMON_IAMCLIENT_TLSCREDENTIALS_HPP_
+#define AOS_COMMON_IAMCLIENT_TLSCREDENTIALS_HPP_
+
+#include <functional>
+#include <memory>
+#include <string>
+
+#include <common/utils/grpchelper.hpp>
+#include <core/common/crypto/itf/certloader.hpp>
+#include <core/common/crypto/itf/x509.hpp>
+#include <core/common/iamclient/itf/certprovider.hpp>
+
+#include "itf/tlscredentials.hpp"
+
+namespace aos::common::iamclient {
+/**
+ * TLS credentials implementation.
+ */
+class TLSCredentials : public TLSCredentialsItf {
+public:
+    /**
+     * Initializes.
+     *
+     * @param mtlsCredentialsFunc MTLS credentials function.
+     * @return Error.
+     */
+    Error Init(const std::string& caCert, aos::iamclient::CertProviderItf& certProvider,
+        crypto::CertLoaderItf& certLoader, crypto::x509::ProviderItf& cryptoProvider);
+
+    /**
+     * Gets MTLS configuration.
+     *
+     * @param certStorage Certificate storage.
+     * @return MTLS credentials.
+     */
+    RetWithError<std::shared_ptr<grpc::ChannelCredentials>> GetMTLSClientCredentials(
+        const String& certStorage) override;
+
+    /**
+     * Gets TLS credentials.
+     *
+     * @return TLS credentials.
+     */
+    RetWithError<std::shared_ptr<grpc::ChannelCredentials>> GetTLSClientCredentials() override;
```

Comment:
What is the difference between GetTLSClientCredentials and GetChannelCredentials?